src/resources/parser/scene.js

```diff
@@ -110,6 +111,21 @@ class SceneParser {
         // Open all children and add them to the node
         len = entityData.children.length;
         const children = entity._children;
+
+        // If the data mis matches the entities created, attempt
+        // to recover
+        let entityChildrenCount = 0;
+        for (let i = 0; i < children.length; i++) {
+            if (children[i] instanceof Entity) {
```

Comment:
can the scene contains anything else other than the Entity? I wonder if this loop is needed at all.

Reply:
Yes, it does. I can't remember exactly what it was from memory (I think graph nodes) but the check did need to be there.